### PR TITLE
Bug/extra inference labels validation

### PR DESCRIPTION
**Pull request by Ben-Epstein**
When someone trains a model, logging data to galileo, we validate that the shape of their labels matches the shape of their logits (for indexing and such).

BUT, if the user comes back the next day and does an inference run over that same run, but with new labels, and new logits, we don't validate that it matches the existing training data logged. This is problematic and will break things in the API. 

This addresses that problem

image

### CODE REVIEW

File: dataquality/core/init.py [hunk 1]
```diff
@@ -89,6 +89,30 @@ def create_log_file_dir(
...
 
+def _set_labels_for_existing_run(project_name: str, run_name: str) -> None:
+    """When a run already exists, fetch and set the labels automatically
+
+    When users call `dq.init` with a project/run that already exists (for example,
+    to run inference), we want to fetch their labels for them and set them as a
+    convenience.
+
+    We also set the `existing_run` attribute to be True, so they cannot change their
+    labels. Their inference labels must match their train/test/val split labels
```
Comment (setu4993) [inline dataquality/core/init.py:100]: Phew!

File: dataquality/core/init.py [hunk 2]
```diff
@@ -89,6 +89,30 @@ def create_log_file_dir(
...
+    """
+    try:
+        labels = dataquality.metrics.get_labels_for_run(project_name, run_name)
+        dataquality.set_labels_for_run(labels)
+        dataquality.get_data_logger().logger_config.existing_run = True
+        print(
+            f"🚀 Found existing run labels. Setting labels for run to {labels}. Do not "
+            f"set labels for this run."
+        )
```
Comment (setu4993) [inline dataquality/core/init.py:109]:
```suggestion
        print(
            f"🚀 Found existing run labels. Setting labels for run to {labels}. You do not "
            f"need to set labels for this run."
        )
```

File: dataquality/clients/api.py [hunk 3]
```diff
@@ -283,10 +283,6 @@ def get_labels_for_run(
         project, run = self._get_project_run_id(
             project_name=project_name, run_name=run_name
         )
-        task_type = self.get_task_type(project, run)
-        if not task and task_type == TaskType.text_multi_label:
-            raise GalileoException("For multi-label runs, a task name must be provided")
```
Comment (setu4993) [inline dataquality/clients/api.py:288]: Why are we removing this?

File: dataquality/core/init.py [hunk 4]
```diff
@@ -89,6 +89,30 @@ def create_log_file_dir(
...
+    # We except all Exceptions here because this should _never_ prohibit the user from
+    # creating a new run.
+    except Exception:
```
Comment (anthonycorletti) [inline dataquality/core/init.py:112]: should we print/ warn that an exception occurred and then continue on?

File: dataquality/core/init.py [hunk 5]
```diff
@@ -191,3 +214,12 @@ def init(
...
     dataquality.get_data_logger().logger_config.reset(factory=True)
+    if not run_created:
+        _set_labels_for_existing_run(project_name, run_name)
+
+
+def delete_run(project_name: str, run_name: str) -> None:
+    """Deletes a run from Galileo"""
+    pid, rid = api_client._get_project_run_id(project_name, run_name)
+    api_client.delete_run(pid, rid)
+    print(f"🗑 Run {project_name}/{run_name} deleted")
```
Comment (setu4993) [inline dataquality/core/init.py:225]: Would this be nicer?

```suggestion
    print(f"🗑 Run {run_name} in project {project_name} deleted.")
```

**Review by anthonycorletti — APPROVED** [review]

**Review by anthonycorletti — APPROVED** [review]

File: dataquality/loggers/logger_config/base_logger_config.py [hunk 6]
```diff
@@ -34,7 +34,8 @@ class BaseLoggerConfig(BaseModel):
     feature_names: List[str] = []
     metadata_documents: Set = set()  # A document is a large str > 1k chars < 10k chars
     finish: Callable = lambda: None  # Overwritten in Semantic Segmentation
-    polygon_idx: int = 0
+    # True when calling `init` with a run that already exists
+    existing_run: bool = False
```
Comment (setu4993) [inline dataquality/loggers/logger_config/base_logger_config.py:38]: 💯

File: dataquality/loggers/logger_config/semantic_segmentation.py [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 
 class SemanticSegmentationLoggerConfig(BaseLoggerConfig):
-    pass
+    polygon_idx: int = 0
```
Comment (setu4993) [inline dataquality/loggers/logger_config/semantic_segmentation.py:5]: Unrelated change?

File: dataquality/core/init.py (hunk 4 above)
Comment (franz101) [inline dataquality/core/init.py:112]: should we log the exception? for example with dq.analytics

**Review by setu4993 — APPROVED** [review]
Amazing. Thanks for the debug the other day and this fix!

File: dataquality/clients/api.py (hunk 3 above)
Comment (Ben-Epstein) [reply dataquality/clients/api.py:288]: because it's actually handled on the API, so this is no longer necessary

File: dataquality/loggers/logger_config/semantic_segmentation.py (hunk 7 above)
Comment (Ben-Epstein) [reply dataquality/loggers/logger_config/semantic_segmentation.py:5]: yea, unrelated but small and easy to do

File: dataquality/clients/api.py (hunk 3 above)
Comment (Ben-Epstein) [reply dataquality/clients/api.py:288]: just noticed it while working through the PR

File: dataquality/core/init.py (hunk 2 above)
Comment (Ben-Epstein) [reply dataquality/core/init.py:109]: nice. THis will break linting, but ill update it locally and push up

File: dataquality/core/init.py (hunk 4 above)
Comment (Ben-Epstein) [reply dataquality/core/init.py:112]: i dont think so. The most common error will just be that there weren't in fact any labels for this run (it never had data logged), and that is OK

**Review by elboy3 — APPROVED** [review]
very nice 👏
